Sources/Testing/Running/Configuration.swift

```diff
@@ -258,6 +258,44 @@ public struct Configuration: Sendable {
 
   /// The test case filter to which test cases should be filtered when run.
   public var testCaseFilter: TestCaseFilter = { _, _ in true }
+
+  // MARK: - Expectation checking
+
+  /// Whether or not reflection of values in expressions checked by expectations
+  /// is enabled.
+  ///
+  /// When the value of this property is `false`, value reflection is disabled
+  /// and values will only be represented using `String(describing:)` instead of
+  /// using `Mirror` to recursively reflect their substructure.
+  public var isValueReflectionEnabled: Bool = true
+
+  /// The maximum number of elements that can included in a single child
+  /// collection when reflecting a value checked by an expectation.
+  ///
+  /// When ``Expression/Value/init(reflecting:)`` is reflecting a value and it
+  /// encounters a child value which is a collection, it consults the value of
+  /// this property and only includes the children of that collection up to this
+  /// maximum count. After this maximum is reached, all subsequent elements are
+  /// omitted and a single placeholder child is added indicating the number of
+  /// elements which have been truncated.
+  public var maximumValueReflectionCollectionCount: Int = 10
+
+  /// The maximum depth of children that can be included in the reflection of a
+  /// checked expectation value.
+  ///
+  /// When ``Expression/Value/init(reflecting:)`` is reflecting a value, it
+  /// recursively reflects that value's children. Before doing so, it consults
+  /// the value of this property to determine the maximum depth of the children
+  /// to include. After this maximum depth is reached, all children at deeper
+  /// levels are omitted and the ``Expression/Value/isTruncated`` property is
+  /// set to `true` to reflect that the reflection is incomplete.
+  ///
+  /// - Note: `Optional` values contribute twice towards this maximum, since
```

Comment:
Perhaps we should special-case optional values, then?

Reply:
I've thought about that before but not sure it's worthwhile in general. I mainly just wanted to call attention to one major consideration for how the default value for this property should be determined.